### PR TITLE
Implement Num.absDiff

### LINKED ISSUE #5270: Implement `Num.absDiff`
Implement the `Num.absDiff` function for Roc

[zulip discussion](https://roc.zulipchat.com/#narrow/stream/304641-ideas/topic/Num.20function.20for.20getting.20the.20difference.20between.20two.20numbers/near/347277847)

[Rust docs for abs_diff](https://doc.rust-lang.org/std/primitive.usize.html#method.abs_diff)

### PR DESCRIPTION
**Pull request by wontem**
Close #5270

- [x] Implement the functionality in `roc`
- [x] Add documentation
- [x] Add happy-path tests
- [x] Add overflow tests
- [ ] ~~Add [LLVM optimization](https://doc.rust-lang.org/src/core/num/uint_macros.rs.html#1641-1644)~~

### CODE REVIEW

File: crates/compiler/test_gen/src/gen_num.rs [hunk 1]
```diff
@@ -3998,3 +3998,49 @@ fn num_count_one_bits() {
...
+    assert_evals_to!(r#"Num.absDiff 2 1"#, 1, i64);
+    assert_evals_to!(r#"Num.absDiff -1 1"#, 2, i64);
+    assert_evals_to!(r#"Num.absDiff 1 -1"#, 2, i64);
+    assert_evals_to!(r#"Num.absDiff Num.minI64 -1"#, i64::MAX, i64);
+
+    assert_evals_to!(r#"Num.absDiff 0.0 0.0"#, 0.0, f64);
+    assert_evals_to!(r#"Num.absDiff 1.0 2.0"#, 1.0, f64);
+    assert_evals_to!(r#"Num.absDiff 2.0 1.0"#, 1.0, f64);
+    assert_evals_to!(r#"Num.absDiff -1.0 1.0"#, 2.0, f64);
+    assert_evals_to!(r#"Num.absDiff 1.0 -1.0"#, 2.0, f64);
```
Comment (folkertdev) [inline crates/compiler/test_gen/src/gen_num.rs:4016]: could you break this up into 2 functions (one for integers, one for floats)? rust runs these asserts sequentially, so this test will take a long time to complete. by breaking it up into smaller pieces, your cores are used more effectively

Comment (wontem) [reply crates/compiler/test_gen/src/gen_num.rs]: Done in https://github.com/roc-lang/roc/pull/5275/commits/54992fb4367777162f171f511ef8d9a3798648cc

File: crates/compiler/test_gen/src/gen_num.rs [hunk 2]
```diff
@@ -3998,3 +3998,70 @@ fn num_count_one_bits() {
...
+    assert_evals_to!(r#"Num.absDiff 0u8 0u8"#, 0, u8);
+    assert_evals_to!(r#"Num.absDiff 1u8 2u8"#, 1, u8);
+    assert_evals_to!(r#"Num.absDiff 2u8 1u8"#, 1, u8);
+    assert_evals_to!(r#"Num.absDiff -1 1"#, 2, i64);
+    assert_evals_to!(r#"Num.absDiff 1 -1"#, 2, i64);
+    assert_evals_to!(r#"Num.absDiff Num.minI64 -1"#, i64::MAX, i64);
+}
+
+#[test]
+#[cfg(any(feature = "gen-llvm", feature = "gen-dev"))]
```
Comment (wontem) [inline crates/compiler/test_gen/src/gen_num.rs:4014]: This ticket is related as the implementation relies on the `NumGt`
https://github.com/roc-lang/roc/issues/2324

**Comment by folkertdev** [comment]
128-bit compares are not yet implemented in the wasm backend. So let's disable the wasm test for 128-bit values too.

**Pull request merged**